### PR TITLE
Ensure so is there for lifecycle

### PR DESCRIPTION
**Pull request by songbell**
### Details:
 - add so handler to hardware tensors from virtual infer requests

### Tickets:
 - 118840

### CODE REVIEW

File: src/plugins/auto/src/infer_request.cpp [hunk 1]
```diff
@@ -56,10 +56,18 @@ ov::auto_plugin::InferRequest::InferRequest(const std::shared_ptr<const ov::auto
...
+            }
+            ov::ISyncInferRequest::set_tensor(input, tensor);
         }
         for (const auto& output : get_outputs()) {
-            ov::ISyncInferRequest::set_tensor(output, m_shared_request->get_tensor(output));
+            auto tensor = m_shared_request->get_tensor(output);
+            if (!tensor._so) {
+                tensor._so = m_shared_request._so;
+            }
+            ov::ISyncInferRequest::set_tensor(output, tensor);
```
Comment (ilya-lavrenov) [inline src/plugins/auto/src/infer_request.cpp:70]: why our Lifetime tests don't catch it?

Comment (songbell) [reply src/plugins/auto/src/infer_request.cpp:70]: 1.0 plugin works fine together with wrapper. when plugin goes to 2.0(e.g GPU), life time tests catches this failure.

Comment (ilyachur) [reply src/plugins/auto/src/infer_request.cpp:70]: @songbell Shouldn't we override get_tensor() default implementations too?

Comment (songbell) [reply src/plugins/auto/src/infer_request.cpp:70]: set_tensor call here already set the tensor with so to m_tensors in base class, get_tensor() will return the correct tensor with so, I don't see the need to override get_tensor() here, is my understanding correct?

Comment (ilyachur) [reply src/plugins/auto/src/infer_request.cpp:70]: Ok, in this case, looks like it will work. 
Did you check this change on GPU 2.0 branch?

Comment (songbell) [reply src/plugins/auto/src/infer_request.cpp:70]: checked on GPU2.0 branch, but found batch failure, updated the PR

**Review by ilyachur — APPROVED** [review]

**Comment by songbell** [comment]
fix a case with 1.0 API + preprocess, in this case when load to BATCH:GPU, ppp will be applied to GPU only, this will result in mismatch of model input/output precision/layout between BATCH and GPU. add temp workaround in BATCH to recover this ppp information when 1.0 API used. please help review again @ilyachur

File: src/plugins/auto_batch/src/plugin.cpp [hunk 2]
```diff
@@ -329,7 +331,58 @@ std::shared_ptr<ov::ICompiledModel> Plugin::compile_model(const std::shared_ptr<
             meta_device.device_batch_size = 1;
         }
     }
-
+    auto ppp_model = model->clone();
+    ov::preprocess::PrePostProcessor preproc(ppp_model);
+    OPENVINO_SUPPRESS_DEPRECATED_START
```
Comment (ilyachur) [inline src/plugins/auto_batch/src/plugin.cpp:336]: Why did you add this code to the auto batch plugin? From my point of view all pre-processing related issues should be already solved on the core level

**Review by ilyachur — COMMENTED** [review]
In auto batch you have a lot of changes, please explain why do we need them.

**Comment by songbell** [comment]
> In auto batch you have a lot of changes, please explain why do we need them.

explain on changes in auto batch, take batch:GPU for example
e.g. batch_infer_request = batch_compiled_model.create_infer_request();

currently in master, with abov create_infer_request call, batch have a chance to return gpu infer infer request directly to app. this can work from lifecycle perspective with gpu 1.0 api + wrappers in core.
since now gpu is migrating to 2.0, get_tensor() is been overrided in gpu plugin as below:
https://github.com/vladimir-paramuzov/openvino/blob/gpu_api2.0/src/plugins/intel_gpu/src/plugin/sync_infer_request.cpp#L295
as you can see above, gpu is operating another tensor map internally, and so is not handled there, and this is reasonable, because for gpu, so can be added later in core api.

In order to resolve the so dependency, as well as decouple the dependency on gpu plugin future implementation changes, I add a simple wrapper in batch to ensure so is correctly handled in batch plugin for the above case.

File: src/plugins/auto_batch/src/plugin.cpp (hunk 2 above)
Comment (songbell) [reply src/plugins/auto_batch/src/plugin.cpp:336]: considering a legacy case:
CNNNetwork net;
net.getInputsInfo().begin()->second->setLayout(Layout::NCHW);
net.getInputsInfo().begin()->second->setPrecision(Precision::U8);
ie.LoadNetwork(net, BATCH:GPU, ...)   // load the network with ppp info through legacy API.

in above case, GPU(no matter 1.0 or 2.0) will finally been operated with new layout/precision, but for batch plugin (or other virtual plugin), the model is still with old layout/precision( because in core we will not apply ppp for virtual plugin).
this will result in mismatch of model input/output information between virtual plugin and gpu plugin. for example, the tensors precision in virtual infer request will be differ with the tensors in gpu infer request. that's why here I add the workaround in batch plugin to restore the legacy ppp info to model. 

do  you have other suggestion maybe in core some api I can leverage to handle this situation? @ilyachur

Comment (ilyachur) [reply src/plugins/auto_batch/src/plugin.cpp:336]: Here we have in the core the transformation which adds the legacy preprocessing...
https://github.com/openvinotoolkit/openvino/blob/master/src/inference/src/dev/core_impl.cpp#L804

From my point of view, only this place should add preprocessing to the plugin and inside the plugin we should avoid any WAs.

From the problem description, as I understand the auto batch should have the same issues like the HETERO plugin, but HETERO works without WAs on the plugin side. So maybe we need to take a look how it works and use the same approach for BATCH?

Comment (songbell) [reply src/plugins/auto_batch/src/plugin.cpp:336]: em.. from my perspective, hetero vs batch are not exactly same regarding tensor sharing.
for hetero, operate on tensors of infer requests can be fully forwarded to underlying hardware infer requests, which means the  middle layer of hetero infer requests can totally be bypassed.
while on batch (or AUTO), it can allocate its own tensors, and then set these tensors to underlying hardware infer requests during workload scheduling. if there is ppp info mismatch between batch model and hardware model, it can be a problem.

Comment (ilyachur) [reply src/plugins/auto_batch/src/plugin.cpp:334]: But, why auto batch cannot use batch model from the plugin? I mean that generate the model on the plugin side and reuse this information?
Or maybe we can add exception for batch plugin. Anyway I would like to avoid the code duplication here.

File: src/plugins/auto_batch/src/compiled_model.cpp [hunk 3]
```diff
@@ -134,26 +134,17 @@ CompiledModel::GetWorkerInferRequest() const {
...
         m_compiled_model_without_batch->create_infer_request(),
         m_compiled_model_without_batch._so};
+    // simpler wrapper if m_compiled_model_with_batch is empty
+    std::shared_ptr<ov::ISyncInferRequest> sync_res = std::make_shared<ov::autobatch_plugin::SyncInferRequest>(
+        std::dynamic_pointer_cast<const ov::autobatch_plugin::CompiledModel>(shared_from_this()),
+        nullptr,
+        0,
+        0);
+    if (m_compiled_model_with_batch)
+        sync_res = create_sync_infer_request();
```
Comment (ilyachur) [inline src/plugins/auto_batch/src/compiled_model.cpp:147]: I think we can avoid creation of infer request 2 times:
```suggestion
    std::shared_ptr<ov::ISyncInferRequest> sync_res;
    if (m_compiled_model_with_batch)
        sync_res = create_sync_infer_request();
    else
        sync_res = std::make_shared<ov::autobatch_plugin::SyncInferRequest>(
        std::dynamic_pointer_cast<const ov::autobatch_plugin::CompiledModel>(shared_from_this()),
        nullptr,
        0,
        0);
```

**Comment by vladimir-paramuzov** [comment]
> smoke_BehaviorTests/OVHoldersTest.LoadedTensor/target_device=GPU

@songbell  Have you checked if it works with proxy plugin disabled?

**Review by p-durandin — APPROVED** [review]

**Review by wangleis — APPROVED** [review]
